test/lib/validation/form/deprecated-appearance.spec.js

```diff
@@ -0,0 +1,169 @@
+      city: 'columns-10'
+    };
+    return deprecatedAppearance.execute({ xformPath, xmlDoc: getXmlDoc(appearances), apiVersion: LATEST_VERSION })
+      .then(output => assertEmpty(output));
+  });
+
+  it(`returns errors for deprecated appearances`, () => {
+    const appearances = {
+      name: 'horizontal',
+      age: 'horizontal-compact',
+      address: 'compact',
+      streetNbr: 'compact-1',
+      city: 'compact-10'
+    };
+    return deprecatedAppearance.execute({ xformPath, xmlDoc: getXmlDoc(appearances), apiVersion: LATEST_VERSION })
+      .then(output => {
+        expect(output.errors).is.empty;
+        expect(output.warnings).to.have.length(6);
+        expect(output.warnings[0]).to.equal(ERROR_HEADER);
+        expect(output.warnings[1]).to.equal(`  - /data/name: replace [horizontal] with [columns]`);
+        expect(output.warnings[2]).to.equal(`  - /data/age: replace [horizontal-compact] with [columns-pack]`);
+        expect(output.warnings[3]).to.equal(`  - /data/address: replace [compact] with [columns-pack no-buttons]`);
+        expect(output.warnings[4]).to.equal(`  - /data/address/street-nbr: replace [compact-1] with [columns-1 no-buttons]`);
+        expect(output.warnings[5]).to.equal(`  - /data/address/city: replace [compact-10] with [columns-10 no-buttons]`);
+      });
+  });
+
+  it(`returns errors for deprecated appearances when mixed with other appearances`, () => {
+    const appearances = {
+      name: 'h1 horizontal',
+      age: 'horizontal-compact compact-1',
+      address: 'field-list compact db-object',
+      streetNbr: 'compact-1 hidden',
+      city: 'compact-10 compact-9 compact-8 compact-7'
+    };
+    return deprecatedAppearance.execute({ xformPath, xmlDoc: getXmlDoc(appearances), apiVersion: LATEST_VERSION })
+      .then(output => {
+        expect(output.errors).is.empty;
+        expect(output.warnings).to.have.length(6);
+        expect(output.warnings[0]).to.equal(ERROR_HEADER);
+        expect(output.warnings[1]).to.equal(`  - /data/name: replace [horizontal] with [columns]`);
+        expect(output.warnings[2]).to.equal(`  - /data/age: replace [horizontal-compact] with [columns-pack]`);
+        expect(output.warnings[3]).to.equal(`  - /data/address: replace [compact] with [columns-pack no-buttons]`);
+        expect(output.warnings[4]).to.equal(`  - /data/address/street-nbr: replace [compact-1] with [columns-1 no-buttons]`);
+        expect(output.warnings[5]).to.equal(`  - /data/address/city: replace [compact-10] with [columns-10 no-buttons]`);
+      });
+  });
+
+  it(`resolves OK for deprecated appearances when no api version is available`, () => {
+    const appearances = {
+      name: 'horizontal',
+      age: 'horizontal-compact',
+      address: 'compact',
+      streetNbr: 'compact-1',
+      city: 'compact-10'
+    };
+    return deprecatedAppearance.execute({ xformPath, xmlDoc: getXmlDoc(appearances) })
+      .then(output => assertEmpty(output));
+  });
+
+  [
```

Comment:
This reminds me TestNG

Reply:
:+1: Yeah I really like parameterized tests!  (Though they do tend to confuse the IDEs...)